src/store/modules/nft.js

```diff
@@ -272,11 +278,38 @@ const actions = {
     commit(TYPES.NFT_SET_NFT_CLASS_PURCHASE_INFO, { classId, info });
     return info;
   },
+  async fetchNFTListingInfo({ commit }, classId) {
+    const res = await this.$api.$get(
+      api.getChainNFTClassListingEndpoint(classId)
+    );
+    const info = res.listings
+      .map(l => {
+        const {
+          class_id: classId,
+          nft_id: nftId,
+          seller,
+          price,
+          expiration,
+        } = l;
+        return {
+          classId,
+          nftId,
+          seller,
+          price: new BigNumber(price).shiftedBy(-9).toNumber(),
+          expiration: new Date(expiration),
+        };
+      })
+      .sort((a, b) => a - b)[0];
+    commit(TYPES.NFT_SET_NFT_CLASS_LISTING_INFO, { classId, info });
+    return info;
+  },
   async lazyGetNFTPurchaseInfo({ getters, dispatch }, classId) {
     let info = getters.getNFTClassPurchaseInfoById(classId);
+    const listingInfo = getters.getNFTClassListingInfoById(classId);
     if (!info) {
       info = await dispatch('fetchNFTPurchaseInfo', classId);
     }
+    if (!listingInfo) dispatch('fetchNFTListingInfo', classId);
     return info;
   },
```

Comment:
Seems the `PurchaseInfo` term implies both the original `PurchaseInfo` and the new `ListingInfo`. Would a new term be considered to differentiate the purchase price from the new price that considers the listing price?

Reply:
`lazyGetNFTPurchaseAndListingInfo`?

Reply:
Yeah, looks good to me